core/src/main/java/org/testcontainers/DockerClientFactory.java

```diff
@@ -115,7 +115,7 @@ public synchronized boolean isDockerAvailable() {
     }
 
     @Synchronized
-    private DockerClientProviderStrategy getOrInitializeStrategy() {
+    private synchronized DockerClientProviderStrategy getOrInitializeStrategy() {
```

Comment:
Already `@Synchronized`

```suggestion
    private DockerClientProviderStrategy getOrInitializeStrategy() {
```